### PR TITLE
deno: 1.15.3 -> 1.16.2

### PR DESCRIPTION
**Pull request by mikroskeem**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mikroskeem** [comment]
Tried enabling tests again, plenty of them fail because they need build output (like mentioned by derivation author) and some of them fail because cache directory creation fails.

Log: https://gist.github.com/mikroskeem/30664296a11091a60b65e71f7429b27a

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 146543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>deno</li>
  </ul>
</details>

File: pkgs/development/web/deno/default.nix [hunk 1]
```diff
@@ -35,20 +35,9 @@ rustPlatform.buildRustPackage rec {
...
-          install -D ${librusty_v8} "target/$v/gn_out/obj/librusty_v8.a"
-        done
-      }
-
-      # Copy over the `librusty_v8.a` file inside target/XYZ/gn_out/obj, symlink not allowed
-      _librusty_v8_setup "debug" "release" "${arch}/release"
-    '';
+  # The v8 package will try to download a `librusty_v8.a` release at build time to our read-only filesystem
+  # To avoid this we pre-download the file and export it via RUSTY_V8_ARCHIVE
+  RUSTY_V8_ARCHIVE = "${librusty_v8}";
```
Comment (06kellyjac) [inline pkgs/development/web/deno/default.nix:40]:
```suggestion
  RUSTY_V8_ARCHIVE = librusty_v8;
```

**Review by 06kellyjac — APPROVED** [review]
Just one suggestion but LGTM

File: pkgs/development/web/deno/default.nix (hunk 1 above)
Comment (mikroskeem) [reply pkgs/development/web/deno/default.nix]: Good catch, thanks.

**Comment by 06kellyjac** [comment]
> Tried enabling tests again, plenty of them fail because they need build output (like mentioned by derivation author) and some of them fail because cache directory creation fails.
> 
> Log: [gist.github.com/mikroskeem/30664296a11091a60b65e71f7429b27a](https://gist.github.com/mikroskeem/30664296a11091a60b65e71f7429b27a)

Thanks for giving the tests a try again.
Yeah I used to maintain a short list of skipped tests but there ended up being more and more breaking each release and it made updating quite a chore

**Review by 06kellyjac — APPROVED** [review]
LGTM

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 146543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>deno</li>
  </ul>
</details>

**Pull request merged**